client/Cargo.toml

```diff
@@ -43,6 +42,7 @@ futures-util = "0.3.25"
 
 
 [dev-dependencies]
+iroha_core = { version = "=2.0.0-pre-rc.13", path = "../core", features = ["dev-telemetry", "telemetry"]}
```

Comment:
> why does it still remain dev dependency?

Because there is still number of things which are depends on core: 
1) `test_network` crate
2) `iroha_core::smartcontracts::wasm` module
3) permissions related stuff (which should be removed by #3106)
4) `iroha_core::genesis` module (we are building custom genesis in `million_accounts_genesis` example)

Reply:
I think `2.` you can resolve in this PR and `3.` will be taken care of.
`1.` and `4.` are both about genesis if I'm not mistaken.

So it's only about genesis? We can fix this I believe, not necessarily in this PR

Reply:
Ok, will return to draft than and see what can be done.

Reply:
arguably, we should have a predefined genesis file for those tests?

Reply:
I think we can and afaik @Arjentix done smt like this in #1618, made integration test use default `genesis.json`.

Reply:
either that or maybe extract that `GenesisNetwork` and stuff to a separate crate?

Reply:
maybe this will be better option because store million account genesis is not really convenient

Reply:
I see what you mean, you could produce it programatically and just paste it there, but it might be very large

Reply:
it might not be bad that you handle this, but I think `transparent_api` will have to be used in integration tests. For instance, `Registrable::build` is only visible when `transparent_api` is active and integration tests rely heavily on being able to construct those. Another option is to make `Registrable` visible always, but that trait is useless for the client

Reply:
IMO this isn't exactly improving the architecture. We want to avoid bidirectional dependency if possible, such that `iroha_client` is slimmed down as far as we can make it go (and then some).

- Genesis is a special case. It's also mostly dead code. Extract into separate translation unit. IDK if there's a way to hint the linker that the code is single-shot, probably automatic. 
- There are circumstances where a separate process is a good choice. Example: `iroha_bootstrap` process that contains config parsing, genesis, bootstrapping and port mapping, which is a large executable with good UI, which hands over to a minimal `irohad` process, that only handles post-genesis  consensus, and responds to API requests. This is not going to save much memory, but it will allow control flow to follow the information flow. 
- Client tests must die. Real integration tests should work with separate processes built in `release` mode, wherever possible. 
- If `Registrable` isn't used, then it shouldn't be visible. However, if this entails adding another feature flag, I'd rather it were visible and unused, than bloat our feature matrix. AFAICT, there may be a way to make it work with `transparent_api`.

Reply:
> There are circumstances where a separate process is a good choice. Example: iroha_bootstrap process that contains config parsing, genesis, bootstrapping and port mapping, which is a large executable with good UI, which hands over to a minimal irohad process, that only handles post-genesis consensus, and responds to API requests. This is not going to save much memory, but it will allow control flow to follow the information flow.

this is an interesting idea

> Client tests must die. Real integration tests should work with separate processes built in release mode, wherever possible.

absolutely

> If Registrable isn't used, then it shouldn't be visible. However, if this entails adding another feature flag, I'd rather it were visible and unused, than bloat our feature matrix. AFAICT, there may be a way to make it work with transparent_api.

no additional feature flags. `Registrable` is not visible unless `transparent_api` is activated

Reply:
ok. So this will be taken care of when we remove integration tests from client?

Reply:
yes, we will depend on `core` until we stop starting iroha peers inside integration tests .

Reply:
maybe you can write a `TODO` in the `client/Cargo.toml` and close this issue

Reply:
excuse me, I meant you can resolve this conversation.